File: trunk/PolarMathCS/PolarMath/Evaluation/Statistics/Average.cs
using PolarMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PolarMath.Evaluation.Statistics
{
    public class Average : IEvaluation<int>
    {
        public int Evaluate(Training training)
        {
            return DoEvaluate( training.Intervals );
        }

        private static int DoEvaluate(ICollection<int> intervals) 
        {
            var intervalsTotal = intervals.Aggregate<int, long>(0, (current, interval) => current + interval);
            return (int) (intervalsTotal / intervals.Count);
        }
    }
}

File: PolarMath/src/org/macondo/polar/evaluation/statistics/Average.java
package org.macondo.polar.evaluation.statistics;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;

import java.util.List;

/**
 * <p></p>
 *
 * Date: 16.03.2008
 *
 * @author <a href="mailto:ktsibriy@gmail.com">Kirill Y. Tsibriy</a>
 */
public class Average implements Evaluation<Integer> {
    public Integer evaluate(Training training) {
        return doEvaluate(training.getIntervals());
    }

    private Integer doEvaluate(List<Integer> intervals) {
        long intervalsTotal = 0;
        
        for (Integer interval : intervals) {
            intervalsTotal += interval;
        }
        return (int) (intervalsTotal / intervals.size());
    }
}
